test/extended/builds/run_policy.go

```diff
@@ -152,7 +152,7 @@ var _ = g.Describe("[Feature:Builds][Slow][Serial] using build configuration run
 						o.Expect(build.Status.CompletionTimestamp).ToNot(o.BeNil(), "completed builds should have a valid completion time")
 						sawCompletion = true
 					}
-					if build.Status.Phase == buildapi.BuildPhaseRunning {
+					if build.Status.Phase == buildapi.BuildPhaseRunning || build.Status.Phase == buildapi.BuildPhasePending {
```

Comment:
do we need to consider builds in new state?  or can we safely assume they will always be transitioned in a timely enough fashion to pending

Reply:
the point of the test is to create a bunch of builds in the new state, then ensure that one at a time they enter pending(or running), with the next one entering pending(or running) when the previous one completes.

considering new would break the test.

Reply:
the point of the change here is that it might take a while for a pod to enter running, which screws up the confirmatoin this test does to say "did this pod enter running sooner after the last build finished".

So instead of needing the pod to enter running, it's sufficient to see that the build entered pending (Which means we created the pod for the build, which we won't do until the previous build completes).   This will be less vulnerable to problem w/ scheduling and container creation which might prevent a pod from entering a running state, despite the build controller doing the right thing.